ecosystem/typescript/sdk/src/plugins/aptos_token.ts

```diff
@@ -0,0 +1,466 @@
+    creator: AptosAccount,
+    token: MaybeHexString,
+    key: string,
+    type: string,
+    value: string,
+    tokenType?: string,
+    extraArgs?: OptionalTransactionArgs,
+  ): Promise<string> {
+    return this.submitTransaction(
+      creator,
+      "add_property",
+      [tokenType || this.tokenType],
+      [HexString.ensure(token).hex(), key, type, getSinglePropertyValueRaw(value, type)],
+      extraArgs,
+    );
+  }
+
+  /**
+   * Remove token property
+   * @param creator Creator account
+   * @param token Token address
+   * @param key the property key stored on-chain
+   * @returns The hash of the transaction submitted to the API
+   */
+  async removeTokenProperty(
+    creator: AptosAccount,
+    token: MaybeHexString,
+    key: string,
+    tokenType?: string,
+    extraArgs?: OptionalTransactionArgs,
+  ): Promise<string> {
+    return this.submitTransaction(
+      creator,
+      "remove_property",
+      [tokenType || this.tokenType],
+      [HexString.ensure(token).hex(), key],
+      extraArgs,
+    );
+  }
+
+  /**
+   * Update token property
+   * @param creator Creator account
+   * @param token Token address
+   * @param key the property key stored on-chain
+   * @param type the property typed stored on-chain
+   * @param value the property value to be stored on-chain
+   * @returns The hash of the transaction submitted to the API
+   */
+  async updateTokenProperty(
+    creator: AptosAccount,
+    token: MaybeHexString,
+    key: string,
+    type: string,
+    value: string,
+    tokenType?: string,
+    extraArgs?: OptionalTransactionArgs,
+  ): Promise<string> {
+    return this.submitTransaction(
+      creator,
+      "update_property",
+      [tokenType || this.tokenType],
+      [HexString.ensure(token).hex(), key, type, getSinglePropertyValueRaw(value, type)],
+      extraArgs,
+    );
+  }
+
+  /**
+   * Transfer a token ownership
+   *
+   * @param owner The account of the current token owner
+   * @param token Token address
+   * @param recipient Recipient address
+   * @returns The hash of the transaction submitted to the API
+   */
+  async transferTokenOwnership(
+    owner: AptosAccount,
+    token: MaybeHexString,
+    recipient: MaybeHexString,
+    tokenType?: string,
+    extraArgs?: OptionalTransactionArgs,
+  ): Promise<string> {
+    const builder = new TransactionBuilderRemoteABI(this.provider.aptosClient, {
+      sender: owner.address(),
+      ...extraArgs,
+    });
+    const rawTxn = await builder.build(
+      "0x1::object::transfer",
+      [tokenType || this.tokenType],
+      [HexString.ensure(token).hex(), HexString.ensure(recipient).hex()],
+    );
+    const bcsTxn = AptosClient.generateBCSTransaction(owner, rawTxn);
+    const pendingTransaction = await this.provider.aptosClient.submitSignedBCSTransaction(bcsTxn);
+    return pendingTransaction.hash;
+  }
+
+  /**
+   * Transfer a token amount from the sender primary_store to the recipient primary_store
```

Comment:
I don't think this really belongs in aptos_token. This is for fungible assets (including fungible tokens), so even for fungible tokens, users should be aware that the way they transfer is same as general fungible assets (i.e. coins)

Reply:
cc @lightmark @gregnazario

Reply:
spoke offline.

Reply:
but that's an interesting question. the closest function we have to transfer a fungible asset is in the `CoinClient`. 
We might be able to leverage that one but 
- Would need to so some modifications for that function so that would introduce a breaking change to the exciting function. 
- Transfer a fungible asset is not the same as transfer a coin, as fungible asset can also be token - so not sure if we should keep that function under `CoinClient`.

Maybe we can have 
- `AptosToken` class - holds functionality for aptos_token
- `FungibleAsset` class - holds functionality for `fungible_asset`
- `FungibleToken` class - inherits from both classes and implements its own functionality

Does that make sense?

Other options can be
- create a new `FungibleClient` - not sure what I think about it

Reply:
We shouldn't need FungibleToken as people can transfer the fungible tokens they have with FungibleAsset client